### PR TITLE
[Form] moved data trimming logic of TrimListener into StringUtil

### PR DESCRIPTION
**Pull request by issei-m**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | yes |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

In the small job which needs not to build a form (e.g. just process a value gotten from Request), I want to reuse the trim logic.

### CODE REVIEW

**Comment by dosten** [comment]
Why put into `StringUtil` and not in `FormUtil`? Also, the `@author` of the `StringUtil` class must be you.

**Comment by issei-m** [comment]
@dosten I also thought so but I decided to put into a new class because FormUtil is singular form. Should this code be moved into FormUtil?

> the `@author` of the StringUtil class must be you.

`TrimListener` was introduced by @webmozart at first. So it would be better to be his name.

File: src/Symfony/Component/Form/Util/StringUtil.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+     * This class should not be instantiated.
+     */
+    private function __construct()
+    {
+    }
+
+    /**
+     * Returns the trimmed data.
+     *
+     * @param mixed $data
```
Comment (Tobion) [inline src/Symfony/Component/Form/Util/StringUtil.php]: it should only accept a string

Comment (stof) [inline src/Symfony/Component/Form/Util/StringUtil.php]: well, this would force to keep the is_string check in TrimListener though

Comment (Tobion) [inline src/Symfony/Component/Form/Util/StringUtil.php]: which is way better in terms of single responsibility.

Comment (Tobion) [inline src/Symfony/Component/Form/Util/StringUtil.php]: why would this method handle non-strings here. doesn't make sense. it's not even documented that non-strings are returned as-is.

Comment (Tobion) [inline src/Symfony/Component/Form/Util/StringUtil.php]: also this will allow us to add scalar type hints later in php 7.

**Comment by issei-m** [comment]
I've just updated PR. Now, StringUtil does not concern argument type.
BTW, it would be better to add my name to `@author` in a patch like this? (I just moved codes, though)

**Comment by issei-m** [comment]
ping @stof @Tobion

File: src/Symfony/Component/Form/Tests/Util/StringUtilTest.php [hunk 2]
```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace Symfony\Component\Form\Tests\Util;
+
+use Symfony\Component\Form\Util\StringUtil;
+
+/**
+ * @author Bernhard Schussek <bschussek@gmail.com>
```
Comment (Tobion) [inline src/Symfony/Component/Form/Tests/Util/StringUtilTest.php]: test classes dont have this annotation

**Comment by issei-m** [comment]
@Tobion fixed it.
And how about this?

> BTW, it would be better to add my name to @author in a patch like this? (I just moved codes, though)

**Comment by xabbuh** [comment]
@issei-m As you just moved code around, keeping @webmozart as the author is the best.

**Comment by issei-m** [comment]
@xabbuh Thanks.

Well then, all issues were resolved.
Ready to be merged?

**Comment by webmozart** [comment]
About the authorship, you should include yourself as author as well. I'd be pretty confused if I'd find myself as only author of a class that I had never seen before ;)

I think this is fine, but it can't be merged into 2.7 as that branch is feature frozen.

**Comment by issei-m** [comment]
@webmozart Thanks for your advice. I'll try to open a new PR soon.

**Pull request closed without merging**